### DISCUSSION TITLE
How to respond with NXDOMAIN to a specific domain only?

### DISCUSSION THREAD

**Discussion by devipasigner** [0xERR0R/blocky#840] [General]
I need this to block use-application-dns.net with NXDOMAIN code rather than null ip otherwise Firefox will still bypass blocky using DOH enabled. Adguard home and pihole already do this by default so I’m wondering how to do this on blocky

**Comment by 0xERR0R** [comment]
I didn't know about this domain in firefox. If I'm getting this right, default case is the domain can be resolved and FF will use DoH. So in your case, you want to block DoH in FF and use default DNS resolver?

**Reply by devipasigner** [reply]
> I didn't know about this domain in firefox. If I'm getting this right, default case is the domain can be resolved and FF will use DoH. So in your case, you want to block DoH in FF and use default DNS resolver?

Yes, if the domain isn’t blocked using NXDOMAIN Firefox will use DoH by default. So I want to block it in my network.

**Reply by devipasigner** [reply]
And I want to do that without blocking other domains using NXDOMAIN since that can cause certain devices to use a fall back dns

**Comment by devipasigner** [comment]
Pihole and adguard home do this by default without any user configuration, so maybe blocky could do the same.

**Comment by devipasigner** [comment]
A NXDOMAIN response is also needed for apples private relay domains, mask.icloud.com and mask-h2.icloud.com
